### PR TITLE
[Metricbeat] haproxy: Add support for stats exposed over HTTPS

### PR DESCRIPTION
**Pull request by mtojek**
This PR enables support for `stats` exposed via HTTPS for `haproxy` module.

Also, dedicated system test using real certs added.

Issue: https://github.com/elastic/beats/issues/14579

### CODE REVIEW

File: metricbeat/module/haproxy/info/info.go [hunk 1]
```diff
@@ -44,17 +45,21 @@ func init() {
 // MetricSet for haproxy info.
 type MetricSet struct {
 	mb.BaseMetricSet
+	*helper.HTTP
 }
 
 // New creates a haproxy info MetricSet.
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
-	return &MetricSet{BaseMetricSet: base}, nil
+	http, err := helper.NewHTTP(base)
```
Comment (jsoriano) [inline metricbeat/module/haproxy/info/info.go]: We should add `settings: ["ssl", "http"]` to `metricbeat/module/haproxy/_meta/fields.yml` so documentation is updated to mention that http and ssl options are supported.

**Comment by jsoriano** [comment]
Btw, would it make sense to add TLS support for the TCP socket too?

File: metricbeat/module/haproxy/_meta/certs/generate.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+openssl genrsa -out server.key 1024
+openssl req -new -key server.key -out server.csr -subj "$SERVER_SUBJECT"
+openssl x509 -req -days 3650 -in server.csr -CA ca.crt -CAkey ca.key -set_serial 01 -out server.crt
+cat server.crt server.key > server.pem
+
+# client certificate creation
+openssl genrsa -out client.key 1024
+openssl req -new -key client.key -out client.csr -subj "$CLIENT_SUBJECT"
+openssl x509 -req -days 3650 -in client.csr -CA ca.crt -CAkey ca.key -set_serial 02 -out client.crt
+openssl x509 -in client.crt -out client.pem -outform PEM
```
[the remark this answers is not in the document]
Comment (mtojek) [reply metricbeat/module/haproxy/_meta/certs/generate.sh]: Regarding Go option - I need these certs to be present in generated docker image and also read by `beats` application spawned in Python tests.

The second script belongs to the filebeat and I'm not quite convinced about creating an inter-beats dependency.

File: metricbeat/module/haproxy/stat/stat.go [hunk 3]
```diff
@@ -44,16 +45,21 @@ func init() {
 // MetricSet for haproxy stats.
 type MetricSet struct {
 	mb.BaseMetricSet
+	*helper.HTTP
 }
 
 // New creates a new haproxy stat MetricSet.
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
-	return &MetricSet{BaseMetricSet: base}, nil
+	http, err := helper.NewHTTP(base)
```
[the remark this answers is not in the document]
Comment (mtojek) [reply metricbeat/module/haproxy/stat/stat.go]: It looks so. It's covered in Python tests.

File: metricbeat/module/haproxy/info/info.go (hunk 1 above)
Comment (mtojek) [reply metricbeat/module/haproxy/info/info.go]: Fixed.

**Comment by mtojek** [comment]
> Btw, would it make sense to add TLS support for the TCP socket too?

Honestly I would wait for a customer asking for such feature, unless you think this is a basic use case.

File: metricbeat/module/haproxy/stat/stat.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsoriano) [reply metricbeat/module/haproxy/stat/stat.go]: Umm, I am worried that we may break the support to some protocol by using something http-specific here. I am not sure if we have tests covering unix sockets for example, and changes in the HTTP helper might unexpectedly break other protocols.
Wdyt about moving this http-specific code to the `haproxy.NewHaproxyClient()` factory?

File: metricbeat/module/haproxy/_meta/certs/generate.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsoriano) [reply metricbeat/module/haproxy/_meta/certs/generate.sh]: Ok, makes sense.

File: metricbeat/module/haproxy/stat/stat.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mtojek) [reply metricbeat/module/haproxy/stat/stat.go]: Good point, it will not break staff in the future.

Fixed!

**Comment by mtojek** [comment]
Jenkins CI errors unrelated, so merging.

**Pull request merged**